edgexfoundry/app-functions-sdk-go
# fix: Properly register metrics for default pipeline and collect processing time

Previous PR registered the pipeline metrics in the wrong location so the default pipeline was missed.
Implementation to collect processing time was missing from previous PR.

Signed-off-by: Leonard Goodell <leonard.goodell@intel.com>







**If your build fails**  due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/app-functions-sdk-go/blob/main/.github/CONTRIBUTING.md

## PR Checklist
Please check if your PR fulfills the following requirements:

      **N/A**

## Testing Instructions
Build and run app template
Verify following in logs
```
level=INFO ts=2022-04-26T00:09:28.2830281Z app=new-app-service source=runtime.go:155 msg="PipelineMessagesProcessed-default-pipeline metric has been registered and wil
l be reported"
level=INFO ts=2022-04-26T00:09:28.2830879Z app=new-app-service source=runtime.go:163 msg="PipelineMessageProcessingTime-default-pipeline metric has been registered and
 will be reported"
```
Verify 8 metric now being reported (instead of 6)
```
level=DEBUG ts=2022-04-26T00:10:14.004752Z app=new-app-service source=reporter.go:165 msg="Publish 8 metrics to the 'edgex/telemetry/new-app-service' base topic"
```

## New Dependency Instructions (If applicable)
